daemons/fenced/fenced_commands.c

```diff
@@ -1954,10 +1955,22 @@ get_capable_devices(const char *host, const char *action, int timeout, bool suic
     while (g_hash_table_iter_next(&gIter, (void **)&key, (void **)&device)) {
         check_type = target_list_type(device);
         if (pcmk__strcase_any_of(check_type, "status", "dynamic-list", NULL)) {
-            devices_needing_async_query++;
-        }
-    }
-
+//            devices_needing_async_query++;
+              int device_timeout = get_action_timeout(device, 
+                                                  pcmk__str_eq(check_type, "status", pcmk__str_casei)? "status" : "list", timeout);
+              if (device_timeout > timeout) {
+                  crm_notice("Since the pcmk_%s_timeout parameter of %s is larger than stonith-timeout(%d), timeout may occur",
+                  pcmk__str_eq(check_type, "status", pcmk__str_casei)? "status" : "list",
+                  device->id, timeout); 
+              } else if ((device_timeout < timeout) && (device_timeout < MIN_QUERY_TIMEOUT)) {
+                  crm_notice("Since the pcmk_%s_timeout parameter of %s is smaller than the minimum timeout(%d) of Query, timeout may occur",
+                  pcmk__str_eq(check_type, "status", pcmk__str_casei)? "status" : "list",
+                  device->id, MIN_QUERY_TIMEOUT); 
+              }
+        }
+    }
```

Comment:
This gives me an idea. Since per_device_timeout is now always the stonith timeout, we can skip this entire iteration here, and do the above timeout comparison in can_fence_host_with_device() instead, before calling schedule_internal_command() for "dynamic-list" or "status".

I think we can skip the MIN_QUERY_TIMEOUT test; if a user configures a very low timeout for list or status, they are familiar with their local environment, and we can assume they know what they're doing.

Reply:
Thanks for your comment.

I think your point is correct, I think it will be simpler and easier to understand.
I agree that the minimum value check is also unnecessary.